### PR TITLE
Fixed: fixed the breadcrumb inconsistency

### LINKED ISSUE #6679: Inconsistency of breadcrumbs detail shown in about page of the website
### URL:

https://nodejs.org/en/about

### Browser Name:

brave

### Browser Version:

1.65.123

### Operating System:

Windows 11

### How to reproduce the issue:

1. the behavior of the breadcrumb is not consistent across the sidebar in about page.
2. the expected behavior should be like the part of the (Project governance) and the (branding of the nodejs) where the visual are expected. 
image

3. rest of the sub parts are only showing about nodejs as the title. the expected behavior is it should be like above mention
4. current behavior problem screenshot:
image

5. expected behavior: 
image

### PR DESCRIPTION
**Pull request by TenzDelek**
Fixes #6679 
## Validation
before:
image
after:
image

### Check List

- [x] I have read the [Contributing Guidelines](https://github.com/nodejs/nodejs.org/blob/main/CONTRIBUTING.md) and made commit messages that follow the guideline.
- [x] I have run `npm run format` to ensure the code follows the style guide.
- [x] I have run `npm run test` to check if all tests are passing.
- [x] I have run `npx turbo build` to check if the website builds without errors.
- [ ] I've covered new added functionality with unit tests if necessary.

### CODE REVIEW

File: components/withBreadcrumbs.tsx [hunk 1]
```diff
@@ -35,9 +35,10 @@ const WithBreadcrumbs: FC = () => {
     // Reduce the pathList to a breadcrumbs array by finding each path in the current navigation layer,
     // updating the currentNode to the found node's items(next layer) for the next iteration.
     return pathList.reduce((breadcrumbs, path) => {
-      const nodeWithCurrentPath = currentNode.find(
-        ([nodePath]) => nodePath === path
-      );
+      const nodeWithCurrentPath = currentNode.find(([nodePath]) => {
+        const matchs = path.toLowerCase().includes(nodePath.toLowerCase());
```
Comment (ovflowd) [inline components/withBreadcrumbs.tsx]: Does a variable need to be assigned here? 🤔 can you just do

```tsx
return path.toLowerCase().includes(nodePath.toLowerCase());
```

?

Comment (TenzDelek) [reply components/withBreadcrumbs.tsx]: @ovflowd  sure will change that, I was debugging with logging the output so forgot to remove that afterward..

**Comment by AugustinMauroy** [comment]
There are still strange behavior https://nodejs-org-git-fork-tenzdelek-fix-breadcrumb-inco-7252f3-openjs.vercel.app/en/learn/getting-started/ecmascript-2015-es6-and-beyond

Capture d’écran 2024-05-04 à 12 52 35

**Comment by TenzDelek** [comment]
> There are still strange behavior https://nodejs-org-git-fork-tenzdelek-fix-breadcrumb-inco-7252f3-openjs.vercel.app/en/learn/getting-started/ecmascript-2015-es6-and-beyond
> 
> Capture d’écran 2024-05-04 à 12 52 35

@AugustinMauroy  thank you for pointing that out, after checking the prior logic i found out that we are not checking the dashtocamel for digit thats why it was acting like that 
before:
image
after:
image

have made the changes in the dashtocamel file..

**Comment by TenzDelek** [comment]
https://github.com/nodejs/nodejs.org/issues/6679#issuecomment-2094119357  @mikeesto @ovflowd @AugustinMauroy  regarding this comment raise, I tried to look through it. I think that the reason why the getInvolved is not getting its breadcrumbs fix,  is the way we are giving the path for the about. currently the about items are its 
image
so that for about section it is working as fine. first getting in the about(True) then goes in its item, then match with the item.
image
### the Error parts starts from here
but for the get Involved the navigation tree is same (about) but when we tried to get in the tree it doesnt work as the item inside the tree is of about and not of get involved. resulting in false in every iteration :
image 

### Solution
#### my thoughts-
- currently there is about only in the tree, there should be
i think that the reason behind it is how we are structuring the object of about, i think that we need to change the path such that  inside the about there should be two object : aboutin and the getinvolved. (check the navigation tree) then we can decide which one we can traverse into about { [aboutin,{its item..}], [getinvolved,{is items..}]
<p style="color: red"> I don't think i will be able to change the object structure as it might break the other component, if a member of node can help with this. it is highly appreciated. 
</span>

**Comment by TenzDelek** [comment]
as the issue is fixed in #6710 , closing this pr

**Pull request closed without merging**